### PR TITLE
Issue 224/google UUID

### LINKED ISSUE #226: array type should not be argument of IsNil
**symptom**
It will panic when a command that include `array` type

**cause of the problem**
I believe this code is the cause of the problem: https://github.com/looplab/eventhorizon/blob/master/command.go#L147

`reflect.IsNil()` defines which types are 'nilable' as:
```
(...)The argument must be a chan, func, interface, map, pointer, or slice value; if it is not, IsNil panics.(...)
```

**solution**
I was able to pass the test by modifying command.go#L147 like this:
```
//...
	case reflect.Map, reflect.Slice:
		return v.IsNil()
	case reflect.Array:
		for i := 0; i < v.Len(); i++ {
			if !isZero(v.Index(i)) {
				return false
			}
		}
		return true
//...
```

Actually, I was working on #224 by myself, and I found this. Because google/uuid is just wrapper of `byte[128]` array type.

I'll send PR solves both #224 and this issue soon.

### PR DESCRIPTION
**Pull request by gongdo**
# Summary
Migrate to use github/google/uuid package instead of UUID package.

# How to test
`go test ./...`

# Issue
It addresses #224
and accepts changes from #226

# Notes
Notice that the Google's `uuid.UUID` is actually an alias of `[16]byte`. It might be a cause of problem if you treat it like string.

### CODE REVIEW

**Comment by gongdo** [comment]
I'm not sure why the Travis CI fails the test:
https://github.com/looplab/eventhorizon/blob/master/examples/todomvc/handler_test.go#L198
It passes in my local dev env(both Ubuntu WSL and Windows).

Do I need to translate uuid(byte array) to string for Mongdb maybe?

**Comment by maxekman** [comment]
Thanks for the PR! Will take a closer look soon. There is definitely something wrong with one of the test cases, let me see if I can fix it.

About the string representation; care must be taken to not break any existing uses, maybe MongoDB is one such case. The logging is also less clear now (check the test in comparison to #225 which fails on the same TC). Maybe something can be done there also?

File: command.go [hunk 1]
```diff
@@ -146,8 +146,15 @@ func isZero(v reflect.Value) bool {
 		// Types that are not allowed at all.
 		// NOTE: Would be better with its own error for this.
 		return true
-	case reflect.Map, reflect.Array, reflect.Slice:
+	case reflect.Map, reflect.Slice:
 		return v.IsNil()
+	case reflect.Array:
```
Comment (maxekman) [inline command.go]: Could you add a test case that triggers the undesired behaviour (which this code then makes the test pass)?

**Comment by maxekman** [comment]
The UUID update is awesome btw, want to get it in soon!

**Comment by gongdo** [comment]
@maxekman Okay. I made #230. It passes CI test too :)

**Comment by maxekman** [comment]
Great work! Let me know when this is rebased and ready for review.

**Comment by gongdo** [comment]
I re-based and merged conflicts. It's messy commit logs for this simple work, isn't it :~)

Please take a look about CI failure. I couldn't reproduce it in my local env.

**Comment by maxekman** [comment]
@gongdo In the meantime, could you please squash the last commit?

**Comment by gongdo** [comment]
@maxekman done.

**Comment by maxekman** [comment]
Fixed in #245.

**Pull request closed without merging**